### PR TITLE
Fixes greylist form vanish

### PR DESCRIPTION
**Pull request by firedranzer**
Fixes the vanishing of greylist form when a negative value was entered. #1910

### CODE REVIEW

**Comment by vstakhov** [comment]
Probably, we should check all thresholds for negative values, not only greylist one.

**Comment by firedranzer** [comment]
@vstakhov is this okay?

File: interface/js/app/config.js [hunk 1]
```diff
@@ -217,17 +217,31 @@ function($) {
...
+                    // String to array for comparison
+                    var eltsArray = JSON.parse(loadActionsFromForm());
+                    if(eltsArray[2]>=0){
+                        rspamd.queryNeighbours('saveactions', null, null, "POST", {}, {
+                            data: elts,
+                            dataType: "json",
+                        });
+                    }
+                    else{
+                        alert("Greylist can not be negative");
```
Comment (moisseev) [inline interface/js/app/config.js]: IMO it's better to use `rspamd.alertMessage()` function.

**Comment by firedranzer** [comment]
@vstakhov @moisseev implemented the changes. please review it.

**Comment by moisseev** [comment]
@firedranzer It's a good practice to reuse the code. Both checks for "Save cluster" and "Save actions" are quite similar. Would you refactor them in a function?

**Comment by firedranzer** [comment]
@moisseev We can only refactor the `else` part because the `Save Cluster` and `Save Actions` have different posting methods.

**Comment by moisseev** [comment]
@firedranzer I think even just `else` part is better than nothing as it will grow when you add a threshold order check, but different methods shouldn't be a problem since you can call them as callback functions.

```js
function yourFunction(callback) {
    callback(args);		
}

yourFunction(queryNeighbours);

yourFunction(queryLocal);
```

**Comment by firedranzer** [comment]
@moissev I think it will be good to refactor just the `else` part because introducing a callback function will unnecessarily complicate the code.

**Comment by moisseev** [comment]
@firedranzer I'm not agree with you. In fact, callback will compact the code, make it easier to read and it is very easy to accomplish.

**Comment by firedranzer** [comment]
@vstakhov @moisseev I have implemented all the changes suggested by you.

**Comment by moisseev** [comment]
LGTM
It would be nice if a native speaker could take a look at alert messages texts.

**Pull request merged**